src/gamma_inc.jl

```diff
@@ -0,0 +1,577 @@
+        t = sm
+        N = 1
+        c=0.0
+        i = trunc(Int,a - 0.5)
+    else
+        rtx = sqrt(x)
+        sm = erfc(rtx)
+        t = exp(-x)/(rtpi*rtx)
+        N=0
+        c=-0.5
+        i = trunc(Int,a - 0.5)
+    end
+    for lp = N : i
+        if i == 0
+            break
+        end
+        c += 1.0
+        t = (x*t)/c
+        sm += t
+    end
+    return 1.0 - sm
+
+end
+# Reference : 'Computation of the incomplete gamma function ratios and their inverse' by Armido R DiDonato , Alfred H Morris.
+# Published in Journal: ACM Transactions on Mathematical Software (TOMS)
+# Volume 12 Issue 4, Dec. 1986 Pages 377-393
+# doi>10.1145/22721.23109
+
+"""
+    gamma_p(a,x,IND)
+    
+    DLMF: https://dlmf.nist.gov/8.2#E4 , https://dlmf.nist.gov/8.2#E5
+    Wiki: https://en.wikipedia.org/wiki/Incomplete_gamma_function
+    IND --> Accuracy desired ; IND=0 means 14 significant digits accuracy , IND=1 means 6 significant digit and IND=2 means only 3 digit accuracy suffices.
+    gamma_p(a,x) or P(a,x) is the Incomplete gamma function ratio given by : ``1/\\Gamma (a) \\int_{0}^{x} e^{-t}t^{a-1} dt``
+"""
+function gamma_p(a::Float64,x::Float64,ind::Integer)
+    iop = ind + 1
+    acc = acc0[iop]
+    if a<0.0 || x<0.0
+        throw(DomainError((a,x,ind,"`a` and `x` must be greater than 0 ---- Domain : (0,inf)")))
+    elseif a==0.0 && x==0.0
+        throw(DomainError((a,x,ind,"`a` and `x` must be greater than 0 ---- Domain : (0,inf)")))
+    elseif a*x==0.0 
+        if x<=a
+            return 0.0
+        else
+            return 1.0
+        end               
+    end
+    
+    if a >= 1.0
+        if a >= big1[iop]
+            @goto l20
+        elseif a > x || x >= x0[iop] || isinteger(2*a)  
+            @goto l30 
+       #else
+       #    return gamma_p_fsum(a,x)
+            
+        end
+    elseif a == 0.5
+        if x >= 0.25
+            return 1.0 - erfc(sqrt(x))
+        end
+        return erf(sqrt(x))
+    elseif x < 1.1
+        return gamma_p_taylor_x(a, x, ind)  
+    end
+    r = rgammax(a,x)
+    if r == 0.0
+        return 1.0
+    else
+        return gamma_p_cf(a, x, ind)    
+    end
+
+    @label l20
+     l = x/a
+     if l == 0.0
+        return 0.0
+     end
+     s = 1.0 - l
+     z = -logmxp1(l)
+     if z >= 700.0/a
+        return 0.0
+     end
+     y = a*z
+     rta = sqrt(a)
+     if abs(s) <= e0[iop]/rta
+        z = sqrt(z + z) 
+        if l < 1.0 
+            z=-z
+        return gamma_p_temme_1(a, x, z, ind)
+        end
+     end
+
+     if abs(s) <= 0.4
+        @goto l200
+     end
+    
+    @label l30
+      r = rgammax(a,x)
+      if r == 0.0
+        if x <= a
+            return 0.0
+        else
+            return 1.0
+        end
+      end 
+      if x <= max(a,alog10)
+        return gamma_p_taylor(a, x, ind)
+      elseif x < x0[iop]
+        return gamma_p_cf(a, x, ind)
+      else
+        return gamma_p_asym(a, x, ind)
+      end
+    
+    @label l200
```

Comment:
since the above block doesn't flow through, this can just be inserted at `@goto l200`

Reply:
yeah, sorry missed it out.